src/Cleanup.jl

```diff
@@ -233,178 +247,212 @@ struct CleanData
-
-    ################################################################
+"""
+Function to process the mValues stored in input_output_matrix:es in the data struct
+TODO: Pack outputs into a single object (preferably a DataFrame)
+"""
+function clean_2d_values(data::Data, split_factor::Float64)
 
     industry_names = data.input_output.industry_names
-
     mapping_105_to_64 = create_map_105_to_64(data.merge_codes_105)
     mapping_64_to_16 = create_map_64_to_16(data.merge_codes_64)
 
-    low_income = select_year(data.household.income.low, year)
-    low_income = combine_dataframe_row_wise(low_income, sum)
+    import_export = clean_matrix(data.input_output.input_output_matrix, industry_names, mapping_105_to_64)
+    import_export = clean_matrix(import_export, names(import_export), mapping_64_to_16)
 
-    high_income = select_year(data.household.income.high, year)
-    high_income = combine_dataframe_row_wise(high_income, sum)
+    imports_import_export = clean_matrix(data.imports.input_output_matrix, industry_names, mapping_105_to_64)
+    imports_import_export = clean_matrix(imports_import_export, names(imports_import_export), mapping_64_to_16)
 
-    capital_current_year = select_year(data.industry.capital, year)
-    mean_capital_current_year = combine_dataframe_row_wise(capital_current_year, mean)
+    eu_import_export = imports_import_export .* split_factor
+    world_import_export = imports_import_export .* (1 - split_factor)
 
-    capital_next_year = select_year(data.industry.capital, year + 1)
-    mean_capital_next_year = combine_dataframe_row_wise(capital_next_year, mean)
+    return import_export, eu_import_export, world_import_export, imports_import_export
 
-    ######################################
-    tax_products = clean_rows(data.others, "Taxes less subsidies on products", industry_names, mapping_105_to_64)
-    tax_production = clean_rows(data.others, "Taxes less subsidies on production", industry_names, mapping_105_to_64)
-    compensation_employees = clean_rows(data.others, "Compensation of employees", industry_names, mapping_105_to_64)
-    gross_operating_surplus_and_mixed_income = clean_rows(data.others, "Gross operating surplus and mixed income", industry_names, mapping_105_to_64)
-    #######################################
+end
 
-    final_consumption = clean_vector(data.input_output.final_consumption, industry_names, mapping_105_to_64)
-    gross_fixed_capital_formation = clean_vector(data.input_output.gross_fixed_capital_formation, industry_names, mapping_105_to_64)
-    delta_v_value_uk = clean_vector(data.input_output.delta_v_value_uk, industry_names, mapping_105_to_64)
+"""
+Function to process the values that are split between uk, eu, world and stored in vectors.
+"""
+function clean_1d_values(val::Vector{Number}, val_imp::Vector{Number}, map_64::Dict{String, String}, map_16::Dict{String, String}, names_105::Vector, names_16::Vector, split_factor::Float64, industries_in_cols::Bool)
```

Comment:
Because of [julia's type system not being invariant covariant/contravariant](https://docs.julialang.org/en/v1/manual/types/#man-parametric-composite-types):
```suggestion
function clean_1d_values(val::Vector{<:Number}, val_imp::Vector{<:Number}, map_64::Dict{String, String}, map_16::Dict{String, String}, names_105::Vector, names_16::Vector, split_factor::Float64, industries_in_cols::Bool)
```

Reply:
Sorry, must have missed this yesterday :disappointed: